src/mlpack/methods/ann/init_rules/kathirvalavakumar_subavathi_init.hpp

```diff
@@ -95,10 +96,12 @@ class KathirvalavakumarSubavathiInitialization
    *
    * @param W Weight matrix to initialize.
    */
-  template<typename eT>
-  void Initialize(arma::Mat<eT>& W)
+  template<typename MatType>
+  void Initialize(MatType& W,
+      const typename std::enable_if_t<IsMatrix<MatType>::value>* = 0)
   {
-    arma::Row<eT> b = s * arma::sqrt(3 / (W.n_rows * dataSum));
+    typedef typename GetRowType<MatType>::type RowType; 
+    RowType b = s * arma::sqrt(3 / (W.n_rows * dataSum));
```

Comment:
I wonder if for most of these implementations we could just call `Initialize(W, W.n_rows, W.n_cols)`.  Up to you if you want to take the effort to do that refactoring... and that assumes that the two functions are the same otherwise.

Reply:
I thought about it, maybe at the very end, because this will get refactored. We need to get rid of `arma::sqrt` and create a new safe function, so will see.